pyproject.toml

```diff
@@ -0,0 +1,114 @@
+[build-system]
+requires = ["setuptools>=78.0"]
```

Comment:
Support for specifying the license with [SPDX expressions](https://spdx.org/licenses/) were added in v77 https://setuptools.pypa.io/en/stable/history.html#v77-0-0.

The [old approaches](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license) will generate warnings in newer setuptool versions.

Also note that v78.0 caused an outrage due to the breaking change for dash-separated keys in setup.cfg which we saw earlier in #12976 (https://github.com/pypa/setuptools/issues/4919). This doesn't affect us if we migrate to pyproject.toml and it's been reverted in v78.0.2, but we might as well ensure our new setup works with the current latest major version.